pkg/visor/logserver/api.go

```diff
@@ -32,114 +27,200 @@ type API struct {
 	}
 }
 
-func (api *API) log(r *http.Request) logrus.FieldLogger {
-	return httputil.GetLogger(r)
-}
-
-func whitelistFSHandler(w http.ResponseWriter, r *http.Request, serveDir string, whitelistedPKs []cipher.PubKey) {
-	start := time.Now()
-	// Get the remote PK.
-	remotePK, _, err := net.SplitHostPort(r.RemoteAddr)
-	if err != nil {
-		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-		return
-	}
-	// Check if the remote PK is whitelisted.
-	whitelisted := false
-	if len(whitelistedPKs) == 0 {
-		whitelisted = true
-	} else {
-		for _, whitelistedPK := range whitelistedPKs {
-			if remotePK == whitelistedPK.String() {
-				whitelisted = true
-				break
+func whitelistAuth(whitelistedPKs []cipher.PubKey) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Get the remote PK.
+		remotePK, _, err := net.SplitHostPort(c.Request.RemoteAddr)
+		if err != nil {
+			c.Writer.WriteHeader(http.StatusInternalServerError)
+			_, err := c.Writer.Write([]byte("500 Internal Server Error"))
+			if err != nil {
+				httputil.GetLogger(c.Request).WithError(err).Errorf("write http response")
 			}
+			c.AbortWithStatus(http.StatusInternalServerError)
+			return
 		}
-	}
-	// If the remote PK is whitelisted, serve the file.
-	if whitelisted {
-		filePath := serveDir + r.URL.Path
-		file, err := os.Open(filePath) //nolint
-		if err != nil {
-			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		// Check if the remote PK is whitelisted.
+		whitelisted := false
+		if len(whitelistedPKs) == 0 {
+			whitelisted = true
+		} else {
+			for _, whitelistedPK := range whitelistedPKs {
+				if remotePK == whitelistedPK.String() {
+					whitelisted = true
+					break
+				}
+			}
+		}
+		if whitelisted {
+			c.Next()
+		} else {
+			// Otherwise, return a 401 Unauthorized error.
+			c.Writer.WriteHeader(http.StatusUnauthorized)
+			_, err := c.Writer.Write([]byte("401 Unauthorized"))
+			if err != nil {
+				httputil.GetLogger(c.Request).WithError(err).Errorf("write http response")
+			}
+			c.AbortWithStatus(http.StatusUnauthorized)
 			return
 		}
-		defer file.Close() //nolint
+	}
+}
 
-		_, filename := path.Split(filePath)
-		w.Header().Set("Content-Type", mime.TypeByExtension(filepath.Ext(filename)))
-		http.ServeContent(w, r, filename, time.Time{}, file)
+func loggingMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		start := time.Now()
+		c.Next()
+		latency := time.Since(start)
+		if latency > time.Minute {
+			latency = latency.Truncate(time.Second)
+		}
+		statusCode := c.Writer.Status()
+		method := c.Request.Method
+		path := c.Request.URL.Path
+		// Get the background color based on the status code
+		statusCodeBackgroundColor := getBackgroundColor(statusCode)
+		// Get the method color
+		methodColor := getMethodColor(method)
+		// Print the logging in a custom format which includes the publickeyfrom c.Request.RemoteAddr ex.:
```

Comment:
Is there a reason we are not using the built in logger here? Seems to be off to use custom logging instead of the Gin or visor logger.

Reply:
Because the default gin logging does not show the public key of the request

the default visor logging is not well suited in this context ; this is much more readable.